amd_openvx_extensions/amd_rpp/source/tensor/Spectrogram.cpp

```diff
@@ -0,0 +1,276 @@
+/*
+Copyright (c) 2019 - 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "internal_publishKernels.h"
+
+struct SpectrogramLocalData {
+    vxRppHandle *handle;
+    Rpp32u deviceType;
+    RppPtr_t pSrc;
+    RppPtr_t pDst;
+    bool centerWindows;
+    bool reflectPadding;
+    RpptSpectrogramLayout spectrogramLayout;
+    Rpp32s power;
+    Rpp32s nfft;
+    Rpp32s windowLength;
+    Rpp32s windowOffset;
+    Rpp32s windowStep;
+    RpptDescPtr pSrcDesc;
+    RpptDescPtr pDstDesc;
+    RpptROI *pSrcRoi;
+    RpptROI *pDstRoi;
+    Rpp32s *pSrcLength;
+    Rpp32f *pWindowFn;
+    size_t inputTensorDims[RPP_MAX_TENSOR_DIMS];
+    size_t outputTensorDims[RPP_MAX_TENSOR_DIMS];
+};
+
+void copy_src_dims_and_update_dst_roi(SpectrogramLocalData *data, RpptROI *src_roi, RpptROI *dst_roi) {
```

Comment:
We are not copying the src_dims here. Just the updation of dst is happening.

Reply:
Please change the name accordinly to update_dst_roi()

Reply:
we are copying src roi values to RPP srcLength function in this function. So i have named this as copy_src_dims_and_update_dst_roi
https://github.com/swetha097/MIVisionX/blob/4445c39f67c2444f022ae7756bb3a0659a79e4da/amd_openvx_extensions/amd_rpp/source/tensor/Spectrogram.cpp#L51